### ISSUE TITLE
"Random" button displays over Wiki page in Mobile Edge (Windows Mobile)

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-html5#250]
There is a display bug with the "Random" button, which remained on screen and visible, covering a small area of Wiki text when scrolling, on Microsoft Edge Mobile (Windows Phone / Mobile). It can be solved by adding: style="position: relative;" to the open <article> tag in index.html, or equivalent in CSS. The "position: relative" declaration seems to ensure that the mobile html interpreter understands the correct stacking for portrait orientation on small screens, and may be a Windows Mobile specific issue.

Mossroy has commented:
> Are we able to reproduce this issue on desktop Edge? (for more easy testing)
> All the UI is based on bootstrap, so maybe this problem has been
> reported to bootstrap, and maybe (I am allowed to dream) this has been
> fixed in a more recent version of bootstrap? Or at least could be
> reported to them (with a simpler test-case)?

**Comment by Jaifroid** [comment]
Unfortunately, despite trying with the Windows Phone emulation mode in Desktop Edge (Developer Tools, Emulation), I can’t reproduce the bug on a desktop browser. So it is Windows Mobile specific. I attach a screenshot of the bug on my Lumia (look at the photo).
![wp_ss_20170616_0002](https://user-images.githubusercontent.com/4304337/27224095-484d1bc4-528b-11e7-8722-20c716712def.png)

It happens “reliably” on every page. Anyway, I’ll have a check to see whether the issue has been reported to bootstrap or fixed there, and if not, will put a pull request for my CSS fix. The fix doesn’t seem to affect other browsers in any way. Well, I’ve tested in Firefox, Edge and Internet Explorer, by launching index.html from the local filesystem on PC, but it would be good to test on other system(s) too.

**Comment by Jaifroid** [comment]
So, I researched bootstrap, and unfortunately the latest stable bootstrap (3.3.7) doesn't fix the issue (I tried it) and doesn't even technically support Edge. Edge support is in bootstrap alpha (4.0.0-alpha), but this version is radically different, doesn't have a default theme, and has dependencies which are cumbersome to install. Therefore the simplest solution is to add the CSS one-liner (in app.css): "position: relative;", which is implemented in https://github.com/Jaifroid/kiwix-html5/commit/86798b2003b4ab24c728ce3c28a8ef6b8be1c0e0 . I've requested this to be pulled.
